lib/routes/shared/security_pin/security_pin_page.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/breez_user_model.dart';
+import 'package:breez/bloc/user_profile/security_model.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/routes/shared/security_pin/prompt_pin_code.dart';
+import 'package:breez/routes/shared/security_pin/security_pin_warning_dialog.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/route.dart';
+import 'package:flutter/material.dart';
+
+class SecurityPage extends StatefulWidget {
+  SecurityPage({Key key}) : super(key: key);
+
+  @override
+  SecurityPageState createState() {
+    return SecurityPageState();
+  }
+}
+
+class SecurityPageState extends State<SecurityPage> {
+  UserProfileBloc _userProfileBloc;
+
+  bool _isInit = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_isInit) {
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      _isInit = true;
+    }
+  }
+
+  void _updateSecurityModel(SecurityModel securityModel, {String pinCode, bool secureBackupWithPin, bool delete = false}) {
+    UpdateSecurityModel updateSecurityModelAction = UpdateSecurityModel(
+        pinCode: (delete) ? null : (pinCode ?? securityModel.pinCode), secureBackupWithPin: (delete) ? false : (secureBackupWithPin ?? securityModel.secureBackupWithPin));
+    _userProfileBloc.userActionsSink.add(updateSecurityModelAction);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    String _title = "Security PIN";
+    return StreamBuilder<BreezUserModel>(
+        stream: _userProfileBloc.userStream,
+        builder: (context, snapshot) {
+          if (!snapshot.hasData) {
+            return Container();
+          } else {
+            return Scaffold(
+              appBar: new AppBar(
+                  iconTheme: theme.appBarIconTheme,
+                  textTheme: theme.appBarTextTheme,
+                  backgroundColor: theme.BreezColors.blue[500],
+                  automaticallyImplyLeading: false,
+                  leading: backBtn.BackButton(),
+                  title: new Text(
+                    _title,
+                    style: theme.appBarTextStyle,
+                  ),
+                  elevation: 0.0),
+              body: ListView(
+                children: _buildSecurityPINTiles(snapshot.data.securityModel),
+              ),
+            );
+          }
+        });
+  }
+
+  List<Widget> _buildSecurityPINTiles(SecurityModel securityModel) {
+    List<Widget> _tiles = List();
+    final _disablePINTile = _buildDisablePINTile(securityModel);
```

Comment:
We don't need the variables here, we can simplify by creating an initial list with the disabledPINTile and then add the rest of the tiles in case `securityModel.pinCode != null`
If variables are needed then without underscore.